securedrop/source_app/decorators.py

```diff
@@ -14,10 +16,10 @@ def login_required(f: Callable) -> Callable:
     @wraps(f)
     def decorated_function(*args: Any, **kwargs: Any) -> Any:
         try:
-            logged_in_source = SessionManager.get_logged_in_user()
+            logged_in_source = SessionManager.get_logged_in_user(db_session=db.session)
```

Comment:
In this PR, I have removed the local cache (due to the problems encountered in #5694) so a DB session now has to be passed to the SessionManager in order to fetch the source record (instead of fetching it from the now-removed cache).